src/App.jsx

```diff
@@ -14,33 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICAL = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const reverseGoods = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
+
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortField(SORT_FIELD_ALPHABETICAL);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortField(SORT_FIELD_LENGTH);
+  };
```

Comment:
```suggestion
  const reverseGoods = () => {
    setIsReversed(!isReversed);
  };

  const sortAlphabetically = () => {
    setSortField(SORT_FIELD_ALPHABETICAL);
  };

  const sortByLength = () => {
    setSortField(SORT_FIELD_LENGTH);
  };
  
  const visibleGoods = getVisibleGoods(goodsFromServer, sortField, isReversed)
```

and reuse that visibleGoods everywhere